### PR TITLE
Use enzyme in all tests

### PR DESCRIPTION
**Pull request by levithomason**
This test removes our old `RenderTree` testing util in favor of `enzyme`.  The remaining tests using it were updated.  Some missing tests were added.  Very minimal component refactoring was made as well.

`shallow` rendering was used everywhere possible.  No test refactoring was done except that necessary to convert the old RenderTree syntax to the new enzyme.

### CODE REVIEW

File: src/addons/Confirm/Confirm.js [hunk 1]
```diff
@@ -14,46 +14,29 @@ export default class Confirm extends Component {
     className: PropTypes.string,
     confirmLabel: PropTypes.string,
     header: PropTypes.string,
-    ref: PropTypes.string,
   }
 
   static defaultProps = {
     abortLabel: 'Cancel',
     confirmLabel: 'Yes',
-    ref: 'modal',
```
Comment (levithomason) [inline src/addons/Confirm/Confirm.js:23]: There is no reason to set an internal access only `ref` name via props.  Anyway, it was not used.

File: src/addons/Confirm/Confirm.js [hunk 2]
```diff
@@ -14,46 +14,29 @@ export default class Confirm extends Component {
...
   }
 
   state = {
     message: 'Are you sure?',
   }
 
-  componentDidMount() {
-    this.deferred = Promise.defer()
```
Comment (levithomason) [inline src/addons/Confirm/Confirm.js:31]: Fixes promise.defer deprecation warning by using `new Promise()` instead.

File: src/elements/Button/Button.js [hunk 3]
```diff
@@ -9,7 +9,6 @@ export default class Button extends Component {
   }
 
   static defaultProps = {
-    children: 'Click Here',
```
Comment (levithomason) [inline src/elements/Button/Button.js:12]: Stardust should not set default content (nor styles).

File: src/modules/Modal/Modal.js [hunk 4]
```diff
@@ -9,14 +9,9 @@ export default class Modal extends Component {
   static propTypes = {
     children: PropTypes.any,
     className: PropTypes.string,
-    ref: PropTypes.string,
     settings: PropTypes.object,
   }
 
-  static defaultProps = {
-    ref: 'modal',
-  }
```
Comment (levithomason) [inline src/modules/Modal/Modal.js:18]: Again, ref props don't make sense.  Also unused.

**Comment by dvdzkwsk** [comment]
👻 , nothing really jumping out to me.

**Pull request merged**